manager/controlapi/service.go

```diff
@@ -239,6 +297,19 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 	}
 
 	var service *api.Service
+	s.store.View(func(tx store.ReadTx) {
+		service = store.GetService(tx, request.ServiceID)
+	})
+	if service == nil {
+		return nil, grpc.Errorf(codes.NotFound, "service %s not found", request.ServiceID)
+	}
+
+	if request.Spec != nil && request.Spec.Endpoint != nil && (!reflect.DeepEqual(request.Spec.Endpoint, service.Endpoint) || !reflect.DeepEqual(request.Spec.Endpoint, service.Spec.Endpoint)) {
```

Comment:
In the case that the original spec was not applied yet and some kind of conflict was found, this would prevent updating the service even though you're not changing anything related to port bindings. I think this would be confusing behavior.

More importantly, the comparison of `request.Spec.Endpoint` to `service.Endpoint` will always fail, because these aren't even the same type. One is `EndpointSpec` and the other is `Endpoint`.